src/agent/rustjail/src/container.rs

```diff
@@ -639,6 +642,37 @@ fn do_init_child(cwfd: RawFd) -> Result<()> {
     do_exec(&args);
 }
 
+// set_stdio_permissions fixes the permissions of PID 1's STDIO
+// within the container to the specified user.
+// The ownership needs to match because it is created outside of
+// the container and needs to be localized.
+fn set_stdio_permissions(uid: libc::uid_t) -> Result<()> {
+    let null_fd = std::fs::File::open("/dev/null")?.as_raw_fd();
+    let fds = [
+        std::io::stdin().as_raw_fd(),
+        std::io::stdout().as_raw_fd(),
+        std::io::stderr().as_raw_fd(),
+    ];
+
+    for fd in &fds {
+        // Skip chown of /dev/null if it was used as one of the STDIO fds.
+        if *fd != null_fd {
```

Comment:
I'd personally invert this to keep the indent minimal:

```rust
 for fd in &fds {
    if *fd == null_fd {
       continue;
   }

   // ...
}
```

Reply:
Done